services/greasyfork/greasyfork-rating.service.js

```diff
@@ -0,0 +1,40 @@
+import { floorCount as floorCountColor } from '../color-formatters.js'
+import { metric } from '../text-formatters.js'
+import BaseGreasyForkService from './greasyfork-base.js'
+
+export default class GreasyForkRatingCount extends BaseGreasyForkService {
+  static category = 'rating'
+  static route = { base: 'greasyfork', pattern: 'rating-count/:scriptId' }
+
+  static examples = [
+    {
+      title: 'Greasy Fork',
+      namedParams: { scriptId: '407466' },
+      staticPreview: this.render({ good: 17, ok: 2, bad: 3 }),
+    },
+  ]
+
+  static defaultBadgeData = { label: 'rating' }
+
+  static render({ good, ok, bad }) {
+    let color = 'lightgrey'
+    const total = good + bad + ok
+    if (total > 0) {
+      const score = (good * 3 + ok * 2 + bad * 1) / total - 1
+      color = floorCountColor(score, 1, 1.5, 2)
+    }
+    return {
+      message: `${metric(good)} good, ${metric(ok)} ok, ${metric(bad)} bad`,
```

Comment:
suppose the only other thing to consider here would be whether or not to include the stats when there's 0 of a type of feedback. for example with the test results combo badge we won't display 0 failed if all the tests passed.

I don't have strong opinions, and would be fine proceeding with this as is, just wanted to gauge your thoughts on whether we should do something similar here

Reply:
I think having all 3 numbers makes the most sense since there are only 3 types of ratings and it shows how many of each type. It would correspond well with the website which shows a number for all categories even if it is 0:

![image](https://user-images.githubusercontent.com/20955511/179434102-2276fa7d-0553-4d27-b207-d834e5d21463.png)

If some categories were left out, it would make it less obvious that the ratings are in distinct categories and for those unfamiliar with the site to know which ones are missing.